design-library/src/components/BccPagination/BccPagination.vue

```diff
@@ -0,0 +1,160 @@
+  displayRightEllipsis?: boolean;
+  align?: "left" | "center" | "right";
+  displayRowsPerPage?: boolean;
+};
+
+type PageNumberOrEllipsis = number | "...";
+
+const props = withDefaults(defineProps<Props>(), {
+  displayLeftEllipsis: true,
+  displayRightEllipsis: true,
+  displayRowsPerPage: true,
+  maxButtonsDisplayed: 3,
+  align: "right",
+  rowsPerPage: 5,
+  rowsPerPageOptions: () => [5, 10, 25, 50],
+});
+const { items, maxButtonsDisplayed, rowsPerPage, displayRightEllipsis, displayLeftEllipsis } =
+  toRefs(props);
+
+const emit = defineEmits(["update:paginatedItems", "update:currentPage", "update:totalPages"]);
+
+const currentPage = ref(1);
+const perPage = ref(rowsPerPage.value.toString());
+const perPageNumber = computed(() => parseInt(perPage.value));
+const totalPages = computed(() => Math.ceil(items.value.length / perPageNumber.value));
+const maxButtons = computed(() => maxButtonsDisplayed.value);
+
+const isFirstPage = computed(() => currentPage.value === 1);
+const isLastPage = computed(() => currentPage.value === totalPages.value);
+
+const paginatedRows = computed(() => {
+  const start = (currentPage.value - 1) * perPageNumber.value;
+  const end = start + perPageNumber.value;
+  return items.value.slice(start, end);
+});
+
+const currentDisplayedPages = computed<PageNumberOrEllipsis[]>(() => {
+  let arr: PageNumberOrEllipsis[] = [];
+
+  let startPage = Math.max(1, currentPage.value - Math.floor(maxButtons.value / 2));
+  let endPage = Math.min(totalPages.value, currentPage.value + Math.floor(maxButtons.value / 2));
+
+  if (endPage - startPage < maxButtons.value) {
+    if (startPage === 1) {
+      endPage = Math.min(totalPages.value, startPage + maxButtons.value - 1);
+    } else if (endPage === totalPages.value) {
+      startPage = Math.max(1, endPage - maxButtons.value + 1);
+    }
+  }
+
+  if (startPage > 1 && displayLeftEllipsis.value === true) {
+    arr.push(1);
+    if (startPage > 2) {
+      arr.push("...");
+    }
+  }
+
+  for (let i: number = startPage; i <= endPage; i++) {
+    arr.push(i);
+  }
+
+  if (endPage < totalPages.value && displayRightEllipsis.value === true) {
+    if (endPage < totalPages.value - 1) {
+      arr.push("...");
+    }
+    arr.push(totalPages.value);
+  }
+
+  return arr;
+});
+
+const goToPage = (numPage: PageNumberOrEllipsis) => {
+  if (numPage === "...") {
+    return;
+  }
+
+  currentPage.value = numPage;
+};
+
+const changePage = (value: number) => {
+  currentPage.value += value;
+  emit("update:currentPage", currentPage.value);
+};
+
+watchEffect(() => {
+  emit("update:paginatedItems", paginatedRows.value);
+  emit("update:totalPages", totalPages.value);
+});
+</script>
+
+<template>
+  <div
+    :class="{
+      'bcc-pagination-left':
+        (align === 'left' && displayRowsPerPage === true) ||
+        (align === 'right' && displayRowsPerPage === false),
+      'bcc-pagination-center': align === 'center',
+      'bcc-pagination': align === 'right',
```

Comment:
Since the underlying CSS is mostly the same, I think it makes more sense to have `bcc-pagination` as a default class and then just adding the `-left` and `-center` variants in addition when applicable. That's how it's done in most other components as well, for example the secondary button is `bcc-button bcc-button-secondary`, with the last class just adding or overriding classes that are different from the default.